GVFS/GVFS.Virtualization/Projection/GitIndexProjection.GitIndexParser.cs

```diff
@@ -105,21 +134,50 @@ private FileSystemTaskResult AddToProjection(GitIndexEntry data)
                 return FileSystemTaskResult.Success;
             }
 
-            private FileSystemTaskResult AddToModifiedFiles(GitIndexEntry data)
+            /// <summary>
+            /// Adjusts the modifed paths and placeholders list for an index entry.
+            /// </summary>
+            /// <param name="gitIndexEntry">Index entry</param>
+            /// <param name="filePlaceholders">
+            /// Dictionary of file placeholders.  AdjustModifedPathsAndPlaceholdersForIndexEntry will
+            /// remove enties from filePlaceholders as they are found in the index.  After
+            /// AdjustModifedPathsAndPlaceholdersForIndexEntry is called for all entries in the index 
+            /// filePlaceholders will contain only those placeholders that are not in the index.
+            /// </param>
+            private FileSystemTaskResult AdjustModifedPathsAndPlaceholdersForIndexEntry(

```

Comment:
Yeah that's as good as anything I would be able to come up with.

Reply:
Thanks for the suggestions.  I went with `AddEntryToModifiedPathsAndRemoveFromPlaceholdersIfNeeded` because it left a little more wiggle room.  

`MoveEntryFromPlaceholderToModifiedPathsIfInIndex` captured one of the things done by this method, but I thought it gave the impression that the placeholder list was the only thing driving if an entry is added to ModifiedPaths (which is not the case).